common/models/user.js

```diff
@@ -428,6 +428,117 @@ module.exports = function(User) {
   };
 
   /**
+   * Default verification options to use when calling User.verify()
+   * This is the full list of possible params, with example values
+   *
+   * ```js
+   * var options = {
+   *      type: 'email',
+   *      mailer: anEmailModel,
+   *      to: 'test@email.com',
+   *      from: 'noreply@email.com'
+   *      subject: 'verification email subject',
+   *      text: 'verification email text',
+   *      headers: {'Mime-Version': '1.0'},
+   *      template: 'path/to/template.ejs',
+   *      redirect: '/',
+   *      verifyHref: '',
+   *      host: 'localhost'
+   *      protocol: 'http'
+   *      port: 3000,
+   *      restApiRoot= '/api',
+   *      tokenGenerator: function (user, cb) { cb("random-token"); }
+   *    };
+   *```
+   *
+   * Usually you should only require to modify a subset of these params
+   * See `User.verify()` and `User.prototype.verify()` doc for params reference
+   */
+
+  User.verifyOptions = {
```

Comment:
Let's discuss what's the best approach for setting up this object. Here are few alternatives I see:

 - `User.verifyOptions` as a property
 - `User.getVerifyOptions()` as a getter function
 - `User.settings.verifyOptions` to allow configuration via model.json

Use cases to consider:
 - How to allow custom models to overwrite inherited settings while preserving selected options from the base class
 - How to make it easy for app developers to provide the options

I think the following may be most flexible:

```js
User.getVerifyOptions() = function() { 
  return this.settings.verifyOptions || {type: 'email'};
}
```

Custom models then can partially overwrite these settings:

```js
MyUser.getVerifyOptions = function() {
  const base = MyUser.base.getVerifyOptions();
  return Object.assign({}, base, {
    // custom values
  });
}
```